### DISCUSSION TITLE
[Bug]: Browsershot node path error again at AWS EC2 instance with Ubuntu OS

### DISCUSSION THREAD

**Discussion by ShaungBhone** [spatie/laravel-pdf#205] [Q&A] [answered]
### What happened?

When clicking the download button to generate a PDF using the Laravel PDF package through Browsershot, the process failed because Puppeteer could not locate the required Chrome installation. The error message states that Chrome (version 129.0.6668.70) could not be found, which suggests that either Puppeteer was not correctly installed or the cache path is misconfigured.![Image](https://github.com/user-attachments/assets/505b2e27-0ad7-4f25-9570-bbe4fb544e3d)

### How to reproduce the bug

I'm not sure if this step is needed or not https://spatie.be/docs/browsershot/v4/requirements#content-installing-puppeteer-on-a-forge-provisioned-server. That installation guide points me to install puppeteer@^17.

### Package Version

^1.5

### PHP Version

^8.2

### Laravel Version

^11.0

### Which operating systems does with happen with?

Linux

### Notes

Thank you so much your kindly help.

**Comment by faneyy** [comment]
Having this exact issue. Anyone got a solution?

**Comment by ShaungBhone** [comment]
Not now. Still bugs.

**Comment by jeevanthelife** [comment]
any updates?
@freekmurze need your atttention here.

**Comment by ShaungBhone** [comment]
You can check in [here](https://discord.com/channels/297040613688475649/1287427708443426908). But for now, whether this is working or not, I haven't checked it out yet. I will check soon.

**Comment by faisal95bd** [comment]
I still have this same error on Amazon Linux 2023 (AL2023) no matter what I try. 

Working fine on the local server, but it totally crapped out on the production server.

**Comment by flatcapco** [comment]
Did anyone find a solution to this - mine is running the latest version on a ubuntu 20 server and has stopped working in production since I last pushed with the same error.

**Reply by flatcapco** [reply]
Please see discussion 188, that fixed my issue along with my comment at the bottom

**Comment by faisal95bd** [comment]
Last month, after trying several methods, I managed to resolve this issue on my production server (Amazon EC2) as follows.
```
        $chromePath = base_path('.cache/puppeteer/chrome/linux-130.0.6723.91/chrome-linux64/chrome');
        $chromeConfigPath = storage_path('app/chrome_config');
        
        if (!is_dir($chromeConfigPath)) {
            mkdir($chromeConfigPath, 0755, true);
        }

        try {
            $pdfContent = Browsershot::html($html)
                ->setChromePath($chromePath)
                ->setEnvironmentOptions([
                    'CHROME_CONFIG_HOME' => $chromeConfigPath,
                ])
                // rest of your code...
```

**Reply by ShaungBhone** [reply]
Still not working. Do I need setIncludePath?
```
The command "PATH=$PATH:/usr/local/bin:/opt/homebrew/bin NODE_PATH=`npm root -g` node '/var/www/weshop/vendor/spatie/browsershot/src/../bin/browser.cjs' '{"url":"file:\/\/\/tmp\/122072259-0179519001734539162\/index.html","action":"pdf","options":{"args":[],"viewport":{"width":800,"height":600},"displayHeaderFooter":false,"margin":
```

**Reply by faisal95bd** [reply]
You can try the setIncludePath method to specify the included path for the Chrome binary.

`->setIncludePath('/usr/bin:/usr/local/bin') // Update this path as needed
`

**Reply by ShaungBhone** [reply]
Still not working.
```
The command "PATH=$PATH:/usr/bin:/usr/local/bin NODE_PATH=`npm root -g` node '/var/www/weshop/vendor/spatie/browsershot/src/../bin/browser.cjs' '{"url":"file:\/\/\/tmp\/694523872-0905444001737088305\/index.html","action":"pdf","options":{"args":["--no-sandbox"],"viewport":{"width":800,"height":600},"displayHeaderFooter":false,"margin":{"top":"32px","right":"32px","bottom":"32px","left":"32px"},"format":"a4","executablePath":"\/var\/www\/weshop\/.cache\/puppeteer\/chrome\/linux-129.0.6668.89\/chrome-linux64\/chrome","env":
```

**Reply by ShaungBhone** [reply]
full error
```
The command "PATH=$PATH:/usr/bin NODE_PATH=`npm root -g` node '/var/www/weshop/vendor/spatie/browsershot/src/../bin/browser.cjs' '{"url":"file:\/\/\/tmp\/997990089-0743617001737090548\/index.html","action":"pdf","options":{"args":["--no-sandbox"],"viewport":{"width":800,"height":600},"displayHeaderFooter":false,"margin":{"top":"32px","right":"32px","bottom":"32px","left":"32px"},"format":"a4","executablePath":"\/var\/www\/weshop\/.cache\/puppeteer\/chrome\/linux-129.0.6668.89\/chrome-linux64\/chrome","env":{"CHROME_CONFIG_HOME":"\/var\/www\/weshop\/storage\/app\/app\/chrome\/.config"},"printBackground":true}}'" failed. Exit Code: 1(General error) Working directory: /var/www/weshop/public Output: ================ Error Output: ================ Error: Browser was not found at the configured executablePath (/var/www/weshop/.cache/puppeteer/chrome/linux-129.0.6668.89/chrome-linux64/chrome) at ChromeLauncher.launch (/var/www/weshop/node_modules/puppeteer-core/lib/cjs/puppeteer/node/BrowserLauncher.js:86:19) at async callChrome (/var/www/weshop/vendor/spatie/browsershot/bin/browser.cjs:94:23)
```

**Comment by heloufir** [comment]
Same issue in my project (local) using Sail on MacOS.
any news?

**Reply by ShaungBhone** [reply]
In local is working for me. Do you need any help, I will help you.

**Comment by himanshumehru** [comment]
Same error. actually on forge ubuntu 24 . any help?

**Comment by himanshumehru** [comment]
Here is the error I'm getting: [2025-01-18 21:58:23] production.ERROR: The command "PATH=$PATH:/usr/local/bin:/opt/homebrew/bin NODE_PATH=`npm root -g` node '/home/test-site/vendor/spatie/browsershot/src/../bin/browser.cjs' '{"url":"file:\/\/\/tmp\/1890090084-0431456001737266297\/index.html","action":"pdf","options":{"path":"\/home\/test-site\/storage\/app\/public\/invoices\/INV-202501185869.pdf","args":["--no-sandbox"],"viewport":{"width":800,"height":600},"displayHeaderFooter":false,"env":{"CHROME_CONFIG_HOME":"\/home\/test-site\/storage\/app\/chrome\/.config"},"printBackground":true}}'" failed.

**Reply by ShaungBhone** [reply]
I give up and using laravel Dom pdf.

**Reply by himanshumehru** [reply]
I'm thinking about doing the same. Can you use tailwind with Dom pdf though?

**Reply by ShaungBhone** [reply]
No, now I'm styling with pure css.

**Comment by amgadkubati** [comment]
You can follow this, it is work for me.

https://stackoverflow.com/questions/75241446/unable-to-get-laravel-browsershot-working-with-puppeteer-ubuntu-20-04-arm-php-8

**Reply by ShaungBhone** [reply]
Thank you let me check.

**Comment by ShaungBhone** [comment, marked as answer]
I got it now. https://gist.github.com/brunoalfred/73f8a0c8dfa431ef0c268e47b70c5854
